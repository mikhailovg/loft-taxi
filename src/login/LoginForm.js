import React from 'react';
import '../App.css';
import {AppConsumer} from '../App';

import {
    Button,
    TextField
} from '@material-ui/core';

export const LoginForm = ({setPage}) => {

    return (
        <AppConsumer>
            {context => {

                const onSubmit = (event) => {
                    event.preventDefault()
                    context.login(event.target.login.value, event.target.password.value)
                }

                return (
                    <div className={'LoginForm'}>
                        <form onSubmit={onSubmit}>
                            <div className={'LoginFormContainer'}>

                                <div className={'LoginFormItem'}>
                                    <h1 className={'LoginTitle'}>Войти</h1>
                                    <div className={'LoginLink'}>
                                        <div>Уже зарегистрированы?</div>
                                        <a onClick={() => setPage('registration')}
                                           className={'Link'}>Зарегистрируйтесь</a>
                                    </div>
                                </div>

                                <TextField
                                    required
                                    label={'Имя пользователя'}
                                    placeholder={'Имя пользователя'}
                                    margin={'normal'}
                                    name={'login'}
                                />

                                <TextField
                                    required
                                    label={'Пароль'}
                                    placeholder={'Пароль'}
                                    margin={'normal'}
                                    name={'password'}
                                    type={'password'}
                                />

                                <Button type={'submit'} variant={'contained'} color={'primary'} className={'LoginButton'}>
                                    Войти
                                </Button>

                            </div>
                        </form>

                    </div>
                )
            }}
        </AppConsumer>
    )
}
