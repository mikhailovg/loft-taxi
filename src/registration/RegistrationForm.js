import React from 'react'
import '../App.css'

import {
    Button,
    TextField
} from '@material-ui/core'
import {Link} from 'react-router-dom'

import {
    getRegister,
    fetchRegisterRequest,
} from '../modules/main'

import {shallowEqual, useSelector, useDispatch} from 'react-redux'
import history from '../history'

export const RegistrationForm = () => {

    const register = useSelector(getRegister, shallowEqual)
    const registerRequest = useSelector(fetchRegisterRequest, shallowEqual)

    const dispatch = useDispatch()

    if (register && register.success && JSON.parse(register.success) === true) {
        localStorage.setItem('registerSuccess', register.success)
        history.push('/map')
    }

    const onSubmit = (event) => {
        event.preventDefault()

        dispatch({
            ...registerRequest,
            payload: {
                email: event.target.login.value,
                password: event.target.password.value,
                name: event.target.name.value,
                surname: event.target.surname.value,
            }
        })
    }

    return (
        <div className={'LoginForm'}>
            <form onSubmit={onSubmit}>
                <div className={'LoginFormContainer'}>
                    <div className={'LoginFormItem'}>
                        <h1 className={'LoginTitle'}>Регистрация</h1>
                        <div className={'LoginLink'}>
                            <div>Уже зарегистрированы?</div>
                            <Link to={'/login'} className={'Link'}>
                                Войти
                            </Link>
                        </div>
                    </div>

                    <TextField
                        required
                        label={'Адрес электронной почты'}
                        placeholder={'Адрес электронной почты'}
                        margin={'normal'}
                        name={'login'}
                    />
                    <TextField
                        required
                        label={'Имя'}
                        placeholder={'Имя'}
                        margin={'normal'}
                        name={'name'}
                    />
                    <TextField
                        required
                        label={'Фамилия'}
                        placeholder={'Фамилия'}
                        margin={'normal'}
                        name={'surname'}
                    />

                    <TextField
                        required
                        label={'Пароль'}
                        placeholder={'Пароль'}
                        margin={'normal'}
                        name={'password'}
                        type={'password'}
                    />

                    <Button type={'submit'} variant={'contained'} color={'primary'}
                            className={'LoginButton'}>
                        Войти
                    </Button>
                </div>
            </form>

        </div>
    )
}