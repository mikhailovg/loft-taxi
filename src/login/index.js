import React from 'react';
import logo from '../logo.svg';
import '../App.css';

import {LoginForm} from "./LoginForm";
import PropTypes from 'prop-types';

export const Login = ({setPage}) => {
    return (
        <div className={'Login'}>
            <div className={'LoginColumn'}>
                <img src={logo} className={'Logo'} alt={'logo'} width={156}/>
            </div>

            <div className={'LoginColumn'}>
                <LoginForm setPage={setPage}/>
            </div>

        </div>
    )
}

Login.propTypes = {
    setPage: PropTypes.func
}
