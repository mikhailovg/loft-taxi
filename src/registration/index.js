import React from 'react';
import logo from '../logo.svg';
import '../App.css';

import {RegistrationForm} from "./RegistrationForm";

export const Registration = ({setPage}) => {
    return (
        <div className="Login">
            <div className="LoginColumn">
                <img src={logo} className="Logo" alt="logo" width={156}/>
            </div>

            <div className="LoginColumn">
                <RegistrationForm setPage={setPage}/>
            </div>

        </div>
    );
}
