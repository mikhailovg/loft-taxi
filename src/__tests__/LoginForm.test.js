import React from 'react';
import {mount, shallow} from 'enzyme';
import '../setupTests'
import renderer from 'react-test-renderer'

import {LoginForm} from '../login/LoginForm'
import {Router} from 'react-router'
import history from '../history'
import {RegistrationForm} from '../registration/RegistrationForm'

describe('Login', () => {
    const wrapper = shallow(<LoginForm/>)
    const routerWrapper = shallow(
        <Router history={history}>
            <RegistrationForm/>
        </Router>
    )

    it('renders LoginForm', () => {
        expect(wrapper.find('.LoginForm').exists()).toEqual(true);
    });

    it('renders correctly', () => {
        const tree = renderer.create(routerWrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

})

// FIX TypeError: window.URL.createObjectURL is not a function
jest.mock('mapbox-gl', () => ({
    Map: () => ({})
}))
jest.mock('react-redux', () => ({
    useDispatch: () => {},
    useSelector: () => ({
        auth: {
            success: true,
            token: 'TOKEN1'
        },
        register: {
            success: true,
            token: 'TOKEN1'
        },
    })
}))