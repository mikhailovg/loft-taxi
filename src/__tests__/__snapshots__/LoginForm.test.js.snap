// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="LoginForm"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="LoginFormContainer"
    >
      <div
        className="LoginFormItem"
      >
        <h1
          className="LoginTitle"
        >
          Войти
        </h1>
        <div
          className="LoginLink"
        >
          <div>
            Уже зарегистрированы?
          </div>
          <a
            className="Link"
            onClick={[Function]}
          >
            Зарегистрируйтесь
          </a>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
        >
          Имя пользователя
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            name="login"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Имя пользователя"
            required={true}
            type="text"
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink={false}
        >
          Пароль
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Пароль"
            required={true}
            type="password"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained LoginButton MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Войти
        </span>
      </button>
    </div>
  </form>
</div>
`;
